MapboxCoreNavigationTests/RouteProgressTests.swift

```diff
@@ -272,4 +272,76 @@ class RouteProgressTests: XCTestCase {
         legProgress.currentStepProgress.distanceTraveled = lineString.distance(to: coordinates[5])! + (lineString.distance()! - lineString.distance(to: coordinates[5])!) / 2.0
         XCTAssertTrue(legProgress.currentSpeedLimit?.value.isInfinite ?? false)
     }
+    
+    func testRouteProggressCodable() {
+        let routeProgress = RouteProgress(route: route, routeIndex: 0, options: routeOptions)
+        
+        let encoder = JSONEncoder()
+        encoder.userInfo[.options] = routeOptions
+        let data = try! encoder.encode(routeProgress)
+        let decoder = JSONDecoder()
+        decoder.userInfo[.options] = routeOptions
+        let decoded = try! decoder.decode(RouteProgress.self, from: data)
+
+        XCTAssertEqual(routeProgress.indexedRoute.0.distance, decoded.indexedRoute.0.distance)
+        XCTAssertEqual(routeProgress.indexedRoute.0.speechLocale, decoded.indexedRoute.0.speechLocale)
+        XCTAssertEqual(routeProgress.indexedRoute.0.shape, decoded.indexedRoute.0.shape)
+        XCTAssertEqual(routeProgress.indexedRoute.0.expectedTravelTime, decoded.indexedRoute.0.expectedTravelTime)
+        XCTAssertEqual(routeProgress.indexedRoute.0.routeIdentifier, decoded.indexedRoute.0.routeIdentifier)
```

Comment:
I could not directly compare original and decoded `Route` due to differences in floating point [de]serialization precision.

Also, I noticed weird behavior: consider a `Waypoint` with a non-nil `name` as a `source` or `destination` of some `RouteLeg`. Decoding `[RouteLeg]` alone works perfectly, while decoding such array as a part of `Route` object - results in `Waypoint.name` to become nil.  I could not find how this happens.

Reply:
> I could not directly compare original and decoded Route due to differences in floating point [de]serialization precision.

In case it helps, MapboxDirectionsTests uses [`JSONSerialization.objectsAreEqual(_:_:approximate:)`](https://github.com/mapbox/mapbox-directions-swift/blob/b7c26f516cdc46637e5a2eb5b775694516f8b7b3/Tests/MapboxDirectionsTests/JSONSerialization.swift#L31) to deep-compare encoded structures while skirting around floating-point imprecision. 

> Also, I noticed weird behavior: consider a `Waypoint` with a non-nil `name` as a `source` or `destination` of some `RouteLeg`. Decoding `[RouteLeg]` alone works perfectly, while decoding such array as a part of `Route` object - results in `Waypoint.name` to become nil.  I could not find how this happens.

It might be related to passing in a RouteOptions without those names in its waypoints. MapboxDirections tries to copy the names from the request to the response, because the Directions API would otherwise put just the snapped street name into this particular field in the response.

Reply:
Yes, I saw `JSONSerialization.objectsAreEqual(_:_:approximate:)` approach but I wanted to compare decoded objects too